core/autoscaling.go

```diff
@@ -346,20 +446,62 @@ func (a *autoScalingGroup) findSpotInstanceRequests() error {
 				},
 			},
 		})
+}
+
+func (a *autoScalingGroup) findSpotInstanceRequests() error {
+
+	resp, err := a.describeSpotInstanceRequests()
 
 	if err != nil {
 		return err
 	}
-	logger.Println("Spot instance requests were previously created for", a.name)
 
-	for _, req := range resp.SpotInstanceRequests {
+	//
+	// Filter out completed spot instance requests
+	//
+	spotRequests := filterOutCompleteSpotInstanceRequests(resp.SpotInstanceRequests)
+
+	//
+	// Filter out cancelled spot instance requests that have timed out, and have no instances
+	//
+	spotRequests = a.filterOutCancelledRequestsWithNoInstances(spotRequests)
+
+	if len(spotRequests) > 0 {
+		logger.Println("Spot instance requests were previously created for", a.name)
+	}
+
+	for _, req := range spotRequests {
 		a.spotInstanceRequests = append(a.spotInstanceRequests,
 			a.loadSpotInstanceRequest(req))
 	}
 
+	// Check if the ASG's desired capacity has been set to 0 inbetween
+	// spot instance creation, and next run to assign spot instance
+	// to the asg
+	a.checkDesiredCapacity()
+
 	return nil
 }
 
+func (a *autoScalingGroup) checkDesiredCapacity() {
+	if *a.Group.DesiredCapacity == 0 && len(a.spotInstanceRequests) > 0 {
+		logger.Println("Desired Capacity is Zero. However, there are Spot Instance Requests.  Proceeding with cancelling them")
+		for _, req := range a.spotInstanceRequests {
```

Comment:
Why not simply call `cancelSIRAndTerminateInstance` here?

Reply:
Have refactored to the code to allow this.